### PR TITLE
add support for rustls

### PR DESCRIPTION
**Pull request by hungl6844**
because openssl is a pain to cross-compile for, I have added the ability for users to switch to another backend instead. since the only configuration required to do so is enabling the features in reqwest, all I had to do was add a feature in ytmapi-rs that tells reqwest to switch the backend.

### CODE REVIEW

**Comment by nick42d** [comment]
Hi @hungl6844 , cheers for the PR! Looks like there is a [GtiHub issue blocking CI](https://github.com/orgs/community/discussions/120966), bear with me while that gets resolved.

**Comment by nick42d** [comment]
Does this approach still use reqwest default features by default? That's my preference for now.

File: ytmapi-rs/Cargo.lock
Comment (nick42d) [inline ytmapi-rs/Cargo.lock:1]: Please delete lockfile, not required for library (my mistake I need to add this to gitignore in future)

File: ytmapi-rs/Cargo.toml [hunk 1]
```diff
@@ -11,7 +11,7 @@ readme = "README.md"
 
 [dependencies]
 tokio = {version = "1.29.1", features = ["full"]}
-reqwest = {version = "0.12.1", features = ["json"]}
+reqwest = {version = "0.12.1", features = ["json"], default-features = false, optional = true}
```
Comment (nick42d) [inline ytmapi-rs/Cargo.toml:14]: Please add reqwest's non-tls related default features (e.g http2) https://docs.rs/reqwest/latest/reqwest/index.html

Comment (nick42d) [inline ytmapi-rs/Cargo.toml:14]: Reqwest is required openssl or not, so should not be optional = true

File: ytmapi-rs/Cargo.toml [hunk 2]
```diff
@@ -22,3 +22,8 @@ chrono = "0.4.31"
 
 [dev-dependencies]
 pretty_assertions = "1"
+
+[features]
```
Comment (nick42d) [inline ytmapi-rs/Cargo.toml:26]: please add "default" feature that uses reqwest defaults (breaks youtui binary otherwise)

**Review by nick42d — CHANGES_REQUESTED** [review]
Added some comments

**Comment by hungl6844** [comment]
First of all, yes, by default, openssl is still used when building the binary, the feature “rusttls-tls” is required to switch away. Thanks for pointing out the issues, I didn’t even realize the binary wasn’t building (I was only using ytmapi-rs, so I never built youtui). I will implement your changes as soon as possible!

**Review by nick42d — APPROVED** [review]

**Comment by nick42d** [comment]
Made some slight changes, if you're happy with that I'll get this merged!

**Comment by hungl6844** [comment]
Go for it!

**Comment by nick42d** [comment]
> Go for it!

We're in - thanks for your contrib :)
In future I'll look to improve this slightly to work around case where multiple features are specified - issue #30 added for tracking.

**Pull request merged**